File: src/main/java/com/cupoftea/discordmc/minecraft/MinecraftChatListener.java
package com.cupoftea.discordmc.minecraft;

import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.AsyncPlayerChatEvent;

import com.cupoftea.discordmc.DiscordMCPlugin;
import com.cupoftea.discordmc.discord.DiscordManager;

public class MinecraftChatListener implements Listener {
    private final DiscordMCPlugin plugin;
    private final DiscordManager discordManager;

    public MinecraftChatListener(DiscordMCPlugin plugin, DiscordManager discordManager) {
        this.plugin = plugin;
        this.discordManager = discordManager;
    }

    @EventHandler
    public void onPlayerChat(AsyncPlayerChatEvent event) {
        Player player = event.getPlayer();
        String message = event.getMessage();
        discordManager.sendMessageToDiscord(player, message);
    }
}

File: src/main/kotlin/com/cupoftea/discordmc/listener/MinecraftChatListener.kt
package com.cupoftea.discordmc.listener

import com.cupoftea.discordmc.bridge.ChatBridgeService
import io.papermc.paper.event.player.AsyncChatEvent
import net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener

class MinecraftChatListener(
    private val chatBridgeService: ChatBridgeService
) : Listener {

    @EventHandler
    fun onPlayerChat(event: AsyncChatEvent) {
        val playerName = event.player.name
        val message = PlainTextComponentSerializer.plainText().serialize(event.message())

        chatBridgeService.sendMinecraftMessage(playerName, message)
    }
}